client/quantum_serverless/core/job.py

```diff
@@ -303,10 +304,30 @@ def logs(self) -> str:
         """Returns logs of the job."""
         return self._job_client.logs(self.job_id)
 
-    def result(self):
-        """Return results of the job."""
+    def result(self, wait=True, cadence=5, verbose=False):
+        """Return results of the job.
+        Args:
+            wait: flag denoting whether to wait for the
+                job result to be populated before returning
+            cadence: time to wait between checking if job has
+                been terminated
+            verbose: flag denoting whether to log a heartbeat
+                while waiting for job result to populate
+        """
+        if wait:
+            if verbose:
+                logging.info("Waiting for job result.")
+            while self._in_terminal_state():
```

Comment:
```suggestion
            while not self._in_terminal_state():
```

Other than that looks good to go :)